Handle gorilla/mux route url bug

jlhawn opened the pull request:
When getting the URL from a v2 registry url builder, it does not
honor the scheme from the endpoint object and will cause an https
endpoint to return urls starting with http.

Docker-DCO-1.1-Signed-off-by: Josh Hawn josh.hawn@docker.com (github: jlhawn)

Commit:
Handle gorilla/mux route url bug

When getting the URL from a v2 registry url builder, it does not
honor the scheme from the endpoint object and will cause an https
endpoint to return urls starting with http.

Docker-DCO-1.1-Signed-off-by: Josh Hawn <josh.hawn@docker.com> (github: jlhawn)

jfrazelle commented:
tests :)

On Fri, Jan 30, 2015 at 5:28 PM, Josh Hawn <notifications@github.com> wrote:

> When getting the URL from a v2 registry url builder, it does not
> honor the scheme from the endpoint object and will cause an https
> endpoint to return urls starting with http.
>
> Docker-DCO-1.1-Signed-off-by: Josh Hawn josh.hawn@docker.com (github:
> jlhawn)
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/docker/docker/pull/10472
> Commit Summary
>
>    - Handle gorilla/mux route url bug
>
> File Changes
>
>    - *M* registry/v2/urls.go
>    <https://github.com/docker/docker/pull/10472/files#diff-0> (12)
>
> Patch Links:
>
>    - https://github.com/docker/docker/pull/10472.patch
>    - https://github.com/docker/docker/pull/10472.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/10472>.
>

unclejack commented:
Has this been fixed upstream?

icecrime commented:
Looks like a relevant issue upstream is gorilla/mux#13 (February 2013), and it sounds like a "won't fix" to me.

jlhawn commented:
@icecrime @unclejack It makes sense that they won't fix it. Since you can specify multiple `Schemes` on the route, it's ambiguous which to use.

Anyway, I've just updated the commit with a better way of building the URLs and added more test cases.

icecrime commented:
LGTM

tiborvass commented:
LGTM

jfrazelle commented:
LGTM
